src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -47,12 +47,19 @@ private void CompilationStartAction(CompilationStartAnalysisContext context)
                 concurrencyLevel: 2, capacity: 0);
 
             context.RegisterSymbolAction(c => SymbolAction(c, idToCachedResult), _symbolKinds);
+            OnCompilationStartAction(context, idToCachedResult);
+        }
+
+        protected virtual void OnCompilationStartAction(
```

Comment:
can you make abstract, so all subclasses have to think about what the right thing to do is.

Reply:
I usually prefer abstract methods but this is sort of like an event... you shouldn't think about it unless you need it. The local function thing is really a special case.

Reply:
hrmm.. i'm on the fence.

Reply:
fixed